test/mock-server.js

```diff
@@ -0,0 +1,88 @@
+'use strict'
+
+const http = require('http')
+
+module.exports = class MockServer {
+  constructor () {
+    this.port = 18181
+    this.server = http.createServer(this.handleRequest.bind(this))
+  }
+
+  getUrl () {
+    return `http://localhost:${this.port}`
+  }
+
+  listen () {
+    this.server.listen(this.port)
+  }
+
+  close (callback) {
+    this.server.close(callback)
+  }
+
+  handleRequest (req, res) {
+    // Check out this micro webframework. Next big thing, I can feel it.
+    return Routes[req.url](req, res)
+  }
+}
+
+let Routes = {
+  '/v1/collect/success': function (req, res) {
+    getBody(req).then((body) => {
+      // Pass request things back so the tests can verify them
+      let out = {status: 'ok', req: {
+        body,
+        method: req.method,
+        headers: req.headers
+      }}
+      res.writeHead(200, { 'Content-Type': 'application/json' })
+      res.end(`${JSON.stringify(out)}\n`)
+    })
+  },
+
+  '/v1/collect/error': function (req, res) {
+    let out = {message: 'bad request'}
+    res.writeHead(400, { 'Content-Type': 'application/json' })
+    res.end(`${JSON.stringify(out)}\n`)
+  },
+
+  '/v1/collect/success/subscribe': function (req, res) {
+    res.writeHead(200, { 'Content-Type': 'text/plain' }) // FIXME: usmanm, is this type correct?
```

Comment:
Not sure if this is the correct content type. It's not strictly json cause no list object wraps the events